Platform/OpenVariableRuntimeDxe/VariableDxe.c

```diff
@@ -684,8 +686,18 @@ VariableServiceInitialize (
   //
   // Avoid setting UEFI 2.x interface member on EFI 1.x.
   //
-  if (SystemTable->RuntimeServices->Hdr.Revision >= EFI_2_00_SYSTEM_TABLE_REVISION) {
-    SystemTable->RuntimeServices->QueryVariableInfo = VariableServiceQueryVariableInfo;
+  // First test all systable elements as some may have been spoofed and pass a single element check
+  // Then check that QueryVariableInfo is specifically available before setting
+  //
+  if (  (SystemTable->Hdr.Revision >> 16 > 1)
```

Comment:
Please use extra braces here and below, i.e. `((SystemTable->Hdr.Revision >> 16U) > 1`